### PR TITLE
E2e/manual regression covering

### PR DESCRIPTION
**Pull request by vlad-dargel**
Covering manual regression with tests:

Verify that user can add Key in RE Cluster DB
Verify that user can add data via CLI in RE Cluster DB
Verify that user can run commands in Workbench in RE Cluster DB
Verify that user can add Key in RE Cloud DB
Verify that user can add data via CLI in RE Cloud DB
Verify that user can run commands in Workbench in RE Cloud DB
Verify that user can add Key in OSS Cluster DB
Verify that user can add data via CLI in OSS Cluster DB
Verify that user can run commands in Workbench in OSS Cluster DB
Verify that user can add Key in Sentinel Primary Group
Verify that user can add data via CLI in Sentinel Primary Group
Verify that user can run commands in Workbench in Sentinel Primary Group
Verify that user can search via Scan more for search pattern and selected data type in OSS Cluster DB
Verify that user can filter per key name using patterns in DB with 10-50 millions of keys
Verify that user can filter per key type in DB with 10-50 millions of keys
Verify that user use Scan More in DB with 10-50 millions of keys (when search by pattern/)
Verify that user can search per exact field name in Hash in DB with 1 million of fields
Verify that user can search per exact element index in List key in DB with 1 million of elements
Verify that user can search per exact member name in Set key in DB with 1 million of members

### CODE REVIEW

**Comment by ofersteinberg** [comment]
@vlad-dargel It is extremely to review with so many files. Please split it into small bulks

**Comment by vlad-dargel** [comment]
> 

sorry, can't do this because all this tests require common changes in helpers
can you review it as it is?
and btw 1/3 of files have very small changes

File: tests/e2e/helpers/keys.ts [hunk 1]
```diff
@@ -42,22 +43,104 @@ export const keyTypes = [
  * @param count The count of keys to add
  * @param keyNameStartWith The name of the key
  */
-export async function populateDBWithHashes(host: string, port: string, count: number, keyNameStartWith: string): Promise<void> {
+export async function populateDBWithHashes(host: string, port: string, keyArguments: AddKeyArguments): Promise<void> {
```
Comment (ofersteinberg) [inline tests/e2e/helpers/keys.ts:46]: Remove `count` from the description since it's not relevant anymore

File: tests/e2e/helpers/keys.ts [hunk 2]
```diff
@@ -42,22 +43,104 @@ export const keyTypes = [
...
     });
-    await client.on('connect', async function() {
-        for (let i = 0; i < count; i++) {
-            const keyName = `${keyNameStartWith}${common.generateWord(20)}`;
-            await client.hset([keyName, 'field1', 'Hello'], async(error: string) => {
-                if (error) {
-                    throw error;
-                }
-            });
+    await client.on('connect', async function () {
```
Comment (ofersteinberg) [inline tests/e2e/helpers/keys.ts:53]: I recommend to close the connection in the end of each function

**Review by ofersteinberg — CHANGES_REQUESTED** [review]
Vlad, you added 18 new tests! I'm sure you could split it into small PRs.
This is definitely not small changes as you mentioned

File: tests/e2e/helpers/database.ts [hunk 3]
```diff
@@ -131,6 +131,20 @@ export async function acceptLicenseTermsAndAddDatabaseApi(databaseParameters: Ad
...
+/**
+ * Accept License terms and add Sentinel database using api
+ * @param databaseParameters The database parameters
+ * @param databaseName The database name
+*/
+export async function acceptLicenseTermsAndAddSentinelDatabaseApi(databaseParameters: SentinelParameters, databaseName: string): Promise<void> {
+    await acceptLicenseTerms();
+    await discoverSentinelDatabaseApi(databaseParameters);
+    // Reload Page to see the database added through api
+    await t.eval(() => location.reload());
```
[the remark this answers is not in the document]
Comment (vlad-dargel) [reply tests/e2e/helpers/database.ts:143]: customer doesn't add database using api request
as I replaced UI steps for preparing test data (including creation/deletion of databases), these tests need to reload page to see added via api request data

File: tests/e2e/helpers/keys.ts (hunk 1 above)
Comment (vlad-dargel) [reply tests/e2e/helpers/keys.ts:46]: fixed

File: tests/e2e/helpers/keys.ts (hunk 2 above)
Comment (vlad-dargel) [reply tests/e2e/helpers/keys.ts:53]: added client.quit(),, thanks!

**Comment by vlad-dargel** [comment]
@ofersteinberg split part of tests into https://github.com/RedisInsight/RedisInsight/pull/1026

**Review by ofersteinberg — APPROVED** [review]

**Pull request merged**